### PR TITLE
T136 extending interface example

### PR DESCRIPTION
**Pull request by niden**

### CODE REVIEW

**Review by danhunsaker — COMMENTED** [review]
I Have Thoughts™

File: en/oop.md [hunk 1]
```diff
@@ -43,6 +43,42 @@ The following class modifiers are supported:
...
+ 	interface MiddlewareInterfaceEx extends MiddlewareInterface
+	{
+	
+	}
+
+From here we can use the "stub" interface throughout our extension. 
+
+	/**
+     * @test
+     * @issue https://github.com/phalcon/zephir/issues/1686
```
Comment (danhunsaker) [inline en/oop.md]: Do we want to include this line in the docs?

File: en/oop.md [hunk 2]
```diff
@@ -43,6 +43,42 @@ The following class modifiers are supported:
     
     }
 
+<a name='classes-interfaces'></a>
+### Implementing Interfaces
+Zephir classes can implement any number of interfaces, provided that these interfaces are visible for the extension to use. However there are times that your extension might require to implement an interface that is built in another extension or package. 
+
+If we want to implement the `MiddlewareInterface` from the `PSR` package, we will need to:
```
Comment (danhunsaker) [inline en/oop.md]: Should probably clarify that we need the PSR extension, specifically, as normal Composer packages won't be available to extensions at load time.

File: en/oop.md [hunk 3]
```diff
@@ -43,6 +43,42 @@ The following class modifiers are supported:
...
+
+	// middlewareinterfaceex.zep
+	namespace Test\Oo\Extend;
+ 	
+ 	use Psr\Http\Server\MiddlewareInterface;
+ 	
+ 	interface MiddlewareInterfaceEx extends MiddlewareInterface
+	{
+	
+	}
```
Comment (danhunsaker) [inline en/oop.md]: Code block?

File: en/oop.md [hunk 4]
```diff
@@ -43,6 +43,42 @@ The following class modifiers are supported:
...
+    {
+        if (!extension_loaded('psr')) {
+            $this->markTestSkipped(
+                "The psr extension is not loaded"
+            );
+        }
+         $this->assertTrue(
+            is_subclass_of(MiddlewareInterfaceEx::class, 'Psr\Http\Server\MiddlewareInterface')
+        );
+    }
```
Comment (danhunsaker) [inline en/oop.md]: Code block?

File: en/oop.md (hunk 1 above)
Comment (niden) [reply en/oop.md]: Removed the @issue

File: en/oop.md (hunk 4 above)
Comment (niden) [reply en/oop.md]: It is tabbed so it gets formatted as a code block

File: en/oop.md (hunk 2 above)
Comment (niden) [reply en/oop.md]: I have a note below stating just that. Do I need to add it here too you think?

File: en/oop.md (hunk 4 above)
Comment (danhunsaker) [reply en/oop.md]: Ah. Doesn't show up as indented here for some reason, and I'm used to fencing code blocks (syntax highlighting FTW?), so I mistook these sections for regular paragraph text. Apologies.

**Review by sergeyklay — CHANGES_REQUESTED** [review]
It seems to me that if we're discussing here about the interfaces provided by php extension, then we should provide the link to it: https://pecl.php.net/package/psr

**Comment by niden** [comment]
@sergeyklay @danhunsaker Changes applied.

**Review by danhunsaker — APPROVED** [review]

**Pull request merged**